sycl/test-e2e/AOT/gpu.cpp

```diff
@@ -0,0 +1,14 @@
+//==--- gpu.cpp - AOT compilation for gen devices using GEN compiler ------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===---------------------------------------------------------------------===//
+
+// REQUIRES: ocloc, gpu
+// UNSUPPORTED: cuda
+// CUDA is not compatible with SPIR.
```

Comment:
I know I'm not supposed to review the code, so this is just food for thought.

Does HIP support SPIR?
Probably we can explicitly list back-ends supporting SPIR in `REQUIRES: ` list and avoid using `UNSUPPORTED`.

Reply:
I would also expect "ocloc" feature not being true for the CUDA backend. I think we're setting it based on the presence of the utility, but, IMO, backend should also be checked for.